src/surface.rs

```diff
@@ -105,6 +105,16 @@ pub fn get_dpi_factor(surface: &wl_surface::WlSurface) -> i32 {
         .dpi_factor
 }
 
+/// Returns the current dpi factor of a surface if it was created by SCTK or falling back to 1 if
+/// it's not.
+pub(crate) fn get_dpi_factor_or_fallback(surface: &wl_surface::WlSurface) -> i32 {
```

Comment:
we can probably name it `try_get_dpi_factor` and make it return `Some(i32)` and handle unwrap from a caller.

Reply:
Sound good :+1: